packages/pigeon/lib/objc_generator.dart

```diff
@@ -608,23 +608,23 @@ void generateObjcHeader(ObjcOptions options, Root root, StringSink sink) {
   indent.writeln('NS_ASSUME_NONNULL_END');
 }
 
-String _dictGetter(
-    List<String> classNames, String dict, NamedType field, String? prefix) {
+String _listGetter(List<String> classNames, String list, NamedType field,
+    int index, String? prefix) {
   if (classNames.contains(field.type.baseName)) {
     String className = field.type.baseName;
     if (prefix != null) {
       className = '$prefix$className';
     }
-    return '[$className nullableFromMap:GetNullableObject($dict, @"${field.name}")]';
+    return '[$className fromList:($list[$index] == [NSNull null] ? nil : $list[$index])]';
```

Comment:
Ah, you just caught a bug I was stuck on. I just duplicated the logic here instead of calling the other function, will fix.

Reply:
I realized why I changed this.

Previously, the Maps empty space contained `nil`, but NSArray cannot contain nil, I had to switch to NSNull instead.

This broke the nullableFromList function which caused me to write a custom fix in the fromList function.
The according to the tests that were already written, the behavior of sending back `nil` instead of an empty class isn't what was wanted anyway, so I've removed this method entirely now.

Let me know your thoughts.